mutex/src/test/workflows.test.ts

```diff
@@ -0,0 +1,108 @@
+import { TestWorkflowEnvironment } from '@temporalio/testing';
+import { WorkflowHandle } from '@temporalio/client';
+import { Runtime, DefaultLogger, Worker } from '@temporalio/worker';
+import { describe, before, after, it } from 'mocha';
+import { currentWorkflowIdQuery, hasLockQuery, lockWorkflow, testLockWorkflow } from '../workflows';
+import assert from 'assert';
+import { v4 as uuidv4 } from 'uuid';
+
+const taskQueue = 'test' + new Date().toLocaleDateString('en-US');
+
+describe('lock workflow', function () {
+  let runPromise: Promise<void>;
+  let worker: Worker;
+  let lockWorkflowHandle: WorkflowHandle<typeof lockWorkflow>;
+  let env: TestWorkflowEnvironment;
+
+  before(async function () {
+    this.timeout(10000);
+    Runtime.install({ logger: new DefaultLogger('WARN') });
+    env = await TestWorkflowEnvironment.create();
+
+    worker = await Worker.create({
+      connection: env.nativeConnection,
+      workflowsPath: require.resolve('../workflows'),
+      taskQueue,
+    });
+
+    runPromise = worker.run();
+  });
+
+  beforeEach(async function () {
+    lockWorkflowHandle = await env.workflowClient.start(lockWorkflow, {
+      taskQueue,
+      workflowId: 'lock-' + uuidv4(),
+    });
+  });
+
+  after(async function () {
+    worker.shutdown();
+    await runPromise;
+
+    await env.teardown();
+  });
+
+  it('handles locking and unlocking', async function () {
+    const testWorkflowId = 'test-' + uuidv4();
+    const testWorkflowHandle = await env.workflowClient.start(testLockWorkflow, {
+      taskQueue,
+      workflowId: testWorkflowId,
+      args: [lockWorkflowHandle.workflowId],
+    });
+
+    let hasLock = await testWorkflowHandle.query(hasLockQuery);
+    assert.ok(!hasLock);
```

Comment:
This seems racy and has a chance of failing on an overloaded machine, you should get rid of this.